### PR TITLE
Add Modal for editing entries visualization settings

### PR DESCRIPTION
**Pull request by samshara**

### CODE REVIEW

File: src/entities/analysisFramework.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    const conditionalWidgets = widgets.filter(widget => widget.widgetId === 'conditionalWidget');
+
+    const values = conditionalWidgets.map((conditional) => {
+        const {
+            properties: { data: { widgets: widgetsInsideConditional = [] } = {} } = {},
+        } = conditional;
+
+        const widgetsWithId = widgetsInsideConditional.map(({ widget }, index) => (
+            {
+                id: `${conditional.id}-${index}`,
```
Comment (tnagorra) [inline src/entities/analysisFramework.js]: I don't think we should use index here!

File: src/entities/analysisFramework.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// eslint-disable-next-line import/prefer-default-export
+export const getAllWidgets = (widgets = []) => {
+    const conditionalWidgets = widgets.filter(widget => widget.widgetId === 'conditionalWidget');
+
+    const values = conditionalWidgets.map((conditional) => {
```
Comment (tnagorra) [inline src/entities/analysisFramework.js]: We can use reduce and avoid using flat()

File: src/redux/selectors/siloDomainData/analysisFramework.js [hunk 3]
```diff
@@ -48,3 +49,30 @@ export const afViewAnalysisFrameworkWidgetsSelector = createSelector(
...
+        const { widgets = [], properties: { statsConfig = {} } = {} } = afView;
+        const allWidgets = getAllWidgets(widgets);
+
+        const selectedWidgets = Object.entries(statsConfig).map(([key, value]) => {
+            const { pk, isConditionalWidget } = value;
+
+            if (isConditionalWidget) {
+                const { selectors } = value;
+                const widgetIndex = selectors[1];
+                const newId = `${pk}-${widgetIndex}`;
```
Comment (tnagorra) [inline src/redux/selectors/siloDomainData/analysisFramework.js]: I don't think we should use index here!

File: src/redux/selectors/siloDomainData/analysisFramework.js [hunk 4]
```diff
@@ -48,3 +49,30 @@ export const afViewAnalysisFrameworkWidgetsSelector = createSelector(
...
+                const conditionalWidget = allWidgets.find(widget => widget.id === newId);
```
Comment (tnagorra) [inline src/redux/selectors/siloDomainData/analysisFramework.js]: What if conditionalWidget is undefined?

File: src/redux/selectors/siloDomainData/analysisFramework.js [hunk 5]
```diff
@@ -48,3 +49,30 @@ export const afViewAnalysisFrameworkWidgetsSelector = createSelector(
...
+    (afView) => {
+        const { widgets = [], properties: { statsConfig = {} } = {} } = afView;
+        const allWidgets = getAllWidgets(widgets);
+
+        const selectedWidgets = Object.entries(statsConfig).map(([key, value]) => {
+            const { pk, isConditionalWidget } = value;
+
+            if (isConditionalWidget) {
+                const { selectors } = value;
+                const widgetIndex = selectors[1];
```
Comment (tnagorra) [inline src/redux/selectors/siloDomainData/analysisFramework.js]: Does selectors always have at least 2 values?

File: src/redux/selectors/siloDomainData/analysisFramework.js [hunk 6]
```diff
@@ -48,3 +49,30 @@ export const afViewAnalysisFrameworkWidgetsSelector = createSelector(
...
+                const newId = `${pk}-${widgetIndex}`;
+                const conditionalWidget = allWidgets.find(widget => widget.id === newId);
+                const { id } = conditionalWidget;
+
+                return { [key]: id };
+            }
+            const widget = allWidgets.find(w => w.id === pk);
+            const { id } = widget;
```
Comment (tnagorra) [inline src/redux/selectors/siloDomainData/analysisFramework.js]: What if widget is not defined?

File: src/views/AnalysisFramework/EditVizSettingsModal/index.js [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+    handleFaramValidationSuccess = (values) => {
+        const {
+            widgets,
+            editFrameworkSettingsRequest,
+        } = this.props;
+
+        const allWidgets = this.getWidgets(widgets);
+
+        const transformedValues = Object.entries(values).map(([key, value]) => {
+            const widget = allWidgets.find(w => w.id === value);
```
Comment (tnagorra) [inline src/views/AnalysisFramework/EditVizSettingsModal/index.js]: What if widget is not defined?

File: src/views/AnalysisFramework/EditVizSettingsModal/index.js [hunk 8]
```diff
@@ -0,0 +1,309 @@
...
+                    },
+                });
+            }
+
+            return ({ [key]: { pk: value } });
+        });
+
+        const properties = {
+            statsConfig: {
+                ...Object.assign({}, ...transformedValues),
```
Comment (tnagorra) [inline src/views/AnalysisFramework/EditVizSettingsModal/index.js]: What does this do?

File: src/views/AnalysisFramework/index.js [hunk 9]
```diff
@@ -204,6 +210,11 @@ export default class AnalysisFramework extends React.PureComponent {
             );
         }
 
+        const {
+            widgets = [],
```
Comment (tnagorra) [inline src/views/AnalysisFramework/index.js]: We shouldn't create new objects in render

File: src/views/AnalysisFramework/index.js [hunk 10]
```diff
@@ -204,6 +210,11 @@ export default class AnalysisFramework extends React.PureComponent {
...
+            properties = {},
```
Comment (tnagorra) [inline src/views/AnalysisFramework/index.js]: We shouldn't create new objects in render

**Review by tnagorra — DISMISSED** [review]
Need to fix some problems first!

**Review by AdityaKhatri — APPROVED** [review]

File: src/views/AnalysisFramework/EditVizSettingsModal/index.js [hunk 11]
```diff
@@ -0,0 +1,303 @@
...
+        const {
+            widgets = [],
+            requests: {
+                editFrameworkSettingsRequest,
+            },
+        } = this.props;
+
+        const allWidgets = this.getWidgets(widgets);
+
+        const transformedValues = Object.entries(values).map(([key, value]) => {
```
Comment (tnagorra) [inline src/views/AnalysisFramework/EditVizSettingsModal/index.js]: Looks too complex: Object.entries().map().reduce()

Comment (tnagorra) [reply src/views/AnalysisFramework/EditVizSettingsModal/index.js]: Could use a mapToMap transformation

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**